### PR TITLE
#167729797 User travel requests history

### PR DESCRIPTION
**Pull request by mthamayor**
#### What does this PR do?
Users can get their travel requests history

#### Description of Task to be completed?
1. Create endpoint *GET api/v1/trips/request?offset=<:offset>&limit=<:limit>*
2. Document endpoint with swagger

#### How should this be manually tested?
After installing dependencies with `npm install`
- Run unit tests with `npm run tests`
- Start the server with `npm run dev`
- Sign up a user or log in an existing user to get user token
- From postman, create a get request *http://localhost:<your port | 3000>/api/v1/trips/request*, enter the `Bearer <token>` in the Authorization header
- You will receive a success response similar to the one in the image below
```
{
    "status": 200,
    "message": "Trips successfully retrieved",
    "data": [
        {
            "tripId": "20c9806e-65d6-4f9a-ba08-463f4256e2bc",
            "tripType": "Multi-city",
            "origin": {},
            "destinations": [
                {},
                {
                    "locationId": 3,
                    "locationName": "Lagos Office 2",
                    "locationAddress": "1233454 Tiamiyu Savage, Victoria Island, Lagos Nigeria"
                }
            ],
            "departureDate": "2019-11-01T00:00:00.000Z",
            "returnDate": "2019-12-10T00:00:00.000Z",
            "reason": "official",
            "requestStatus": "pending",
            "createdAt": "2019-09-06T09:48:54.227Z"
        }
    ],
    "pagination": {
        "limit": "1",
        "offset": 0,
        "totalCount": 1
    }
}
```

#### Any background context you want to provide?
- Please check the [implementation plan](https://docs.google.com/document/d/1ZTOHYHDVzirpfz1X15CWrym54MOwRKsXFeYoiC2f7ls/edit?usp=sharing) for `response` specifications
- There is a switch in Swagger versioning from swagger 2.0 to openapi 3.0.1 to enable access to use `Bearer <token>` for protected endpoints

#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/167729797

#### Screenshots (if appropriate)
Proof of passing tests
Screenshot (189)

#### Questions:
None

### CODE REVIEW

**Review by sebaztyn — CHANGES_REQUESTED** [review]
Hey @mthamayor, I ran the endpoint on postman. It worked great for a user with an existing history.
When I signed up a new user and ran the request. Its return message was not clear as seen in the attached screenshot.
A message such as "No trip request found" or something in that form might be appropriate.
image

File: src/test/getRequests.test.js [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+          expect(res).to.have.property('status');
+          expect(res).to.have.property('body');
+          expect(res.status).to.deep.equal(200);
+
+          const { body } = res;
+
+          expect(body).to.have.property('status');
+          expect(body).to.have.property('data');
+
+          const { status, data } = body;
```
Comment (sebaztyn) [inline src/test/getRequests.test.js]: I think you forgot to take down that console.log on line 124

File: src/test/getRequests.test.js [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+          console.log(data);
```
Comment (sebaztyn) [inline src/test/getRequests.test.js]: I think you forgot to take down that console.log on line 124

**Review by sebaztyn — CHANGES_REQUESTED** [review]

**Review by sebaztyn — APPROVED** [review]

**Review by okabamac — APPROVED** [review]

File: .eslintrc.json [hunk 3]
```diff
@@ -11,11 +11,10 @@
     "arrow-parens": 0,
     "one-var-declaration-per-line": 0,
     "new-cap": 0,
-    "arrow-parens": 0, 
     "consistent-return": 0,
     "no-param-reassign": 0,
     "comma-dangle": 0,
-    "arrow-parens": 0,
+    "no-await-in-loop": "off",
```
Comment (Stylll) [inline .eslintrc.json]: This Eslint rule is turned off. It's a good practice not to use await inside a loop. You should turn this rule on.

File: src/controllers/userRequestHistory.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
+import models from '../models';
```
Comment (Stylll) [inline src/controllers/userRequestHistory.js]: Can we have this controller inside the Trips.js file instead? That file contains the methods to create trips so it'll be better if we have the method to get user trips in it also.

**Review by Stylll — CHANGES_REQUESTED** [review]
@mthamayor, Good work so far. Just a few comments. Please review.

**Review by Stylll — CHANGES_REQUESTED** [review]
hey, @mthamayor I noticed there's no message in the response. Can we have a message saying something like this `Trips successfully retrieved` or `No trips available` if the user has no trips?.
@sebaztyn Pointed that out in his comments but it wasn't fixed.

**Review by Stylll — CHANGES_REQUESTED** [review]
Hey @mthamayor I noticed when one of the locations in a request destination array does not exist in the DB, (ie, in the requests table the destination column has two locations: 3, 989). 3 exists but 989 does not exist. This is the response that is returned

```
{
    "status": "error",
    "error": "a location does not exist"
}
```
Ideally, it should just return an empty object for that destination.

**Review by sebaztyn — APPROVED** [review]
I have tested locally and on postman. This get request works effectively.
Thank you.

**Review by davidfrank96 — APPROVED** [review]
Pulled it and it works fine on my machine

**Review by Stylll — APPROVED** [review]

**Pull request merged**